pkg/display/manifest_printer.go

```diff
@@ -0,0 +1,212 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package display
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// TreeFormat is used to format tree based output using 4 values.
+// Each value must display with the same total width to format correctly.
+//
+// MiddleDrop is used to show a child element which is not the last child
+// LastDrop is used to show the last child element
+// SkipLine is used for displaying data from a previous child before the next child
+// Spacer is used to display child data for the last child
+type TreeFormat struct {
+	MiddleDrop string
+	LastDrop   string
+	SkipLine   string
+	Spacer     string
+}
+
+// LineTreeFormat uses line drawing characters to format a tree
+//
+// TreeRoot
+// ├── First child       # MiddleDrop =  "├── "
+// │   Skipped line      # SkipLine = "│   "
+// └── Last child        # LastDrop = "└── "
+// ....└── Only child    # Spacer="....", LastDrop = "└── "
+var LineTreeFormat = TreeFormat{
+	MiddleDrop: "├── ",
+	LastDrop:   "└── ",
+	SkipLine:   "│   ",
+	Spacer:     "    ",
+}
+
+type ContentReader interface {
+	content.Provider
+
+	Info(ctx context.Context, dgst digest.Digest) (content.Info, error)
+}
+
+type Printer struct {
```

Comment:
Should be called something like `ImageTreePrinter`, or moved to a subpackage like `pkg/display/imagetree`?